### PR TITLE
Add table of API objects to README.md

### LINKED ISSUE #113: Better overview on existing/missing Foreman API modules
The following Foreman API resources exist according to https://theforeman.org/api/1.17/

- [ ] Architectures
- [x] ~~Audits~~ - readonly
- [ ] Auth source ldaps
- [ ] Autosign
- [ ] Bookmarks
- [x] Common parameters - [foreman_global_parameter](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_global_parameter.py)
- [ ] Compute attributes
- [x] Compute profiles - [foreman_compute_profile](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_compute_profile.py)
- [x] Compute resources - [foreman_compute_resource](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_compute_resource.py)
- [ ] Config groups
- [ ] Config reports
- [ ] Config templates
- [x] ~~Dashboard~~ - readonly
- [x] Domains - [foreman_domain](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_domain.py)
- [ ] Environments
- [ ] External usergroups
- [x] ~~Fact values~~ - readonly
- [ ] Filters
- [x] ~~Home~~ - readonly
- [ ] Host classes
- [ ] Hostgroup classes
- [ ] Hostgroups
- [ ] Hosts
- [ ] Http proxies
- [ ] Images
- [ ] Interfaces
- [x] Locations - [foreman_location](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_location.py)
- [x] ~~Mail notifications~~ - readonly
- [ ] Media
- [ ] Models
- [x] Operating systems - [foreman_operating_system](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_operating_system.py)
- [x] Organizations - [foreman_organization](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_organization.py)
- [x] Os default templates - [foreman_os_default_template](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_os_default_template.py)
- [ ] Override values
- [ ] Parameters
- [x] ~~Permissions~~ - readonly
- [ ] Personal access tokens
- [x] ~~Plugins~~ - readonly
- [x] Provisioning templates - [foreman_provisioning_template](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_provisioning_template.py)
- [x] Ptables - [foreman_ptable](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_ptable.py)
- [ ] Puppet hosts
- [ ] Puppetclasses
- [x] Realms
- [ ] Reports
- [ ] Roles
- [x] Settings
- [ ] Smart class parameters
- [ ] Smart proxies
- [ ] Smart variables
- [ ] Ssh keys
- [x] ~~Statistics~~ - readonly
- [ ] Subnets
- [x] ~~Tasks~~ - readonly
- [ ] Template combinations
- [x] ~~Template kinds~~ - readonly
- [ ] Usergroups
- [ ] Users

Not all of these are really useful/interesting for Ansible modules. Some only support GET parameters, so they might be interesting for extracting facts or variables or asserting on their content, but they can't really be configured for example.

Still some modules are missing for my use case (manage/store the configuration of a Foreman instance in Ansible). It would be nice to have a list like the above for Foreman and Katello also in the README.md file to get a quick overview what's already possible and what not.

I can open a PR for this, if you like.

### PR DESCRIPTION
**Pull request by MarkusTeufelberger**
This might give a better overview of which modules exist and which are still needed.

fixes #113

### CODE REVIEW

**Comment by mdellweg** [comment]
Now, it looks more like a TODO/ROADMAP list. We could put it in a corresponding file. In the README i would prefer to state what we have, rather than what we do not.
Also as this is the first time (except briefly here https://github.com/theforeman/foreman-ansible-modules/pull/62#issuecomment-342517986), an actual version of foreman/katello has been mentioned, i realised we never stated, that we want to support a specific version.

**Comment by MarkusTeufelberger** [comment]
Ansible versions supported are also not really explicitly mentioned, though I would hope that this repository gets moved upstream soon (e.g. for 2.6) anyways. The "Action modules" would need to be moved into existing ones or rewritten to be idempotent, other than that I don't see any big reason why to keep them together here.

A list like this could also help to refer to already upstreamed modules while this repo serves as "incubator" or a s a place to find modules.

**Comment by akofink** [comment]
I agree it makes sense that the readme should state what is currently possible. I'd be fine with putting this information in a separate file.

**Comment by mdellweg** [comment]
I am happy to see this list here somewhere, just not in the README.
Maybe take the 'Goals'-section and put them together with this list in ROADMAP (or FEATURE_MATRIX or you_name_it).
Then link to it from the README.
Further i think, we should move the available modules list up in the README, and put the testing stuff in a 'Contributing'-Section (or separate file).
The README is really the first impression someone gets from a project, and i think by now we might have more users than contributors visiting this.

BTW: 'Global Parameter' is how it is called in the gui. Is there a nice way to have an alias for whole modules? Both names are equally valid.

**Comment by akofink** [comment]
Sometimes projects will create a `docs/` folder (i.e. https://github.com/theforeman/forklift/tree/master/docs). Perhaps we should have a `testing.md`,  `contributing.md`, `roadmap.md` etc. and leave the Readme as a place for new users of the project as @mdellweg said.

**Comment by MarkusTeufelberger** [comment]
To be honest, I'd rather see this repository go away as fast as possible with all modules available in Ansible upstream than starting to build more structure into it. There are already a bunch of modules that would be very useful to have upstream and that I would consider complete enough to make it there (and also be maintained and tested).

It was already a bit of research to find this repository and to make it useable it in my projects (this one is not the only one that makes Foreman API objects available in Ansible and I'm not even sure if the approach with this "nailgun" library instead of just using the REST API directly is a good design decision, especially if you take a deeper look into nailgun code). It is really easy to have a community maintained module upstream so you can review and merge code - https://groups.google.com/forum/#!topic/foreman-users/alDRxJiWJGs sounded as well like you want to move forward faster half a year ago.

As a relative new user to this repository, I assumed that this is mostly a "move fast and get things upstream" incubator, not a fully fledged project with roadmaps etc. Knowing what's already available and what not would have been helpful, especially considering the poor state of the foreman and katello module upstream (which are worse than if they didn't exist actually).

As roadmaps have to come from the contributors (you know better which modules are planned to be done next), feel free to take the list from this PR and do the necessary changes. I personally think it should be in README and it is something that people will want to know and read about these modules - maybe even more than how to test them or debug Ansible modules.

**Comment by akofink** [comment]
We should definitely start talking about submitting PRs to Ansible. We can always change/add production modules later with additional Ansible PRs. Even having a handful of feature complete modules in Ansible would increase the visibility of this project.

**Comment by ehelms** [comment]
I broke this merging another PR. Please rebase!

File: README.md [hunk 1]
```diff
@@ -89,27 +89,124 @@ You can set a number of environment variables besides `MODULE` to configure make
...
+- [ ] Roles
+- [ ] Settings
+- [ ] Smart class parameters
+- [ ] Smart proxies
+- [ ] Smart variables
+- [ ] Ssh keys
+- [ ] Statistics - readonly
+- [ ] Subnets
+- [ ] Subscriptions
+- [ ] Sync - readonly
```
Comment (sean797) [inline README.md]: Isn't this covered by `katello_sync` under `Action Modules` ?

**Comment by ehelms** [comment]
@MarkusTeufelberger Could you please rebase this?

File: README.md [hunk 2]
```diff
@@ -89,27 +89,124 @@ You can set a number of environment variables besides `MODULE` to configure make
...
+- [x] Common parameters - [foreman_global_parameter](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_global_parameter.py)
+- [ ] Compute attributes
+- [x] Compute profiles - [foreman_compute_profile](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_compute_profile.py)
+- [x] Compute resources - [foreman_compute_resource](https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/foreman_compute_resource.py)
+- [ ] Config groups
+- [ ] Config reports
+- [ ] Config templates
+- [ ] Containers
+- [ ] Content uploads
+- [ ] Content view components
```
[the remark this answers is not in the document]
Comment (akofink) [reply README.md]: Part of https://github.com/theforeman/foreman-ansible-modules/pull/177

**Comment by MarkusTeufelberger** [comment]
Closing, since there still doesn't seem to be consensus on where this information should end up (README.md/Roadmap.md/...?) ~1 year after writing and researching this initially. Also I don't want to contribute to (or use) modules outside of Ansible upstream, so please don't merge, fix or rebase this PR.

**Pull request closed without merging**